x-pack/platform/plugins/shared/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -608,7 +608,18 @@ The Kibana Connector in use may need to be reconfigured with an updated Amazon B
       connectorUsageCollector
     );
 
-    return response.data.pipe(new PassThrough()) as unknown as IncomingMessage;
+    if (response.data) {
+      const resultStream = response.data as SmithyMessageDecoderStream<unknown>;
+      // splits the stream in two, [stream = consumer, tokenStream = token tracking]
+      const [stream, tokenStream] = tee(resultStream);
+      return {
+        ...response.data,
+        stream: Readable.from(stream).pipe(new PassThrough()),
+        tokenStream: Readable.from(tokenStream).pipe(new PassThrough()),
+      };
+    }
+
+    return { ...response.data };
```

Comment:
I'd assume response.data is null or undefined in line 622, if that's the case response.data is not destructible.

Reply:
https://github.com/elastic/kibana/pull/227531/commits/20dcf210c77452a7a2d40e7a40a0391b011cb717